fairlearn/postprocessing/_threshold_optimizer.py

```diff
@@ -114,25 +111,25 @@ def fit(self, X, y, *, sensitive_features, **kwargs):
         else:
             y = [int(y_val) for y_val in y]
 
-        if self._estimator:
+        if self.estimator:
             # train estimator on data first
-            self._validate_estimator()
-            self._estimator.fit(X, y, **kwargs)
-            self._unconstrained_predictor = self._estimator
+            self.estimator_ = self.estimator.fit(X, y, **kwargs)
```

Comment:
This is mostly for my own understanding: doesn't this mean that `self.estimator` and `self.estimator_` are the same thing?

Reply:
Oh I guess we did the same thing before as well... nvm

Reply:
yeah this is what was done before. It should probably be instead cloning before fitting, but didn't wanna include that in this PR.